internal/recommender/metrics/provider.go

```diff
@@ -0,0 +1,100 @@
+package metrics
+
+import (
+	"context"
+	"fmt"
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/api"
+	"github.com/prometheus/client_golang/api/prometheus/v1"
+	"github.com/prometheus/common/model"
+	"github.com/scylladb/go-log"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"net/url"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"time"
+)
+
+const (
+	appLabel       = "prometheus"
+	componentLabel = "server"
+	releaseLabel   = "scylla-prom"
+)
+
+type (
+	Provider interface {
+		GetRackMetrics(ctx context.Context, scName string, dcName string, rackName string) (*RackMetrics, error)
```

Comment:
It's good that you defined interface, what I suggest is to generalize it a bit. Perhaps user would want to configure autoscaling behavior based on his external metrics not tied to rack nor dc. If we for example use such interface:
```
type Provider interface {
GetMetric(ctx context.Context, name string, labels []string) (Metric, error)
}
```
Then you could have implementaiton which would translate `scName`, `dcName`, `rackName` into `name` and `labels` and interface would be much generic, and same thing could be accomplished, but it would be easier to provide different sources of metrics. WDYT?